x-pack/auditbeat/module/system/host/_meta/fields.yml

```diff
@@ -95,3 +95,36 @@
         type: ip
         description: >
           IP addresses.
+  - name: users
+    type: array
+    description: >
+        `users` contains information about the users on this system. See the man page for passwd(5) for more detailed information.
+    fields:
+    - name: name
+      type: keyword
+      description: >
+          User name.
+    - name: uid
+      type: integer
+      description: >
+          User ID.
+    - name: gid
+      type: integer
+      description: >
+          Group ID.
+    - name: passwd
+      type: keyword
+      description: >
+          Information about a user's password.
+    - name: dir
```

Comment:
I think I would start with `keyword` for `dir` and `shell` then add an analyzed version later if requested.